app/repositories/sift/ApplicationSiftRepository.scala

```diff
@@ -0,0 +1,131 @@
+import model._
+import model.persisted.{ PassmarkEvaluation, SchemeEvaluationResult }
+import reactivemongo.api.DB
+import reactivemongo.bson.{ BSONArray, BSONDocument, BSONObjectID }
+import repositories.application.GeneralApplicationRepoBSONReader
+import repositories.{ CollectionNames, CommonBSONDocuments, RandomSelection, ReactiveRepositoryHelpers }
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+trait ApplicationSiftRepository extends RandomSelection with ReactiveRepositoryHelpers with GeneralApplicationRepoBSONReader {
+  this: ReactiveRepository[_, _] =>
+
+  def thisApplicationStatus: ApplicationStatus
+  def dateTime: DateTimeFactory
+  def siftableSchemes: Seq[Scheme]
+  val phaseName = "SIFT_PHASE"
+
+  def nextApplicationsForSiftStage(maxBatchSize: Int): Future[List[ApplicationForSift]]
+  def progressApplicationToSiftStage(application: ApplicationForSift): Future[Unit]
+
+  def findApplicationsReadyForSchemeSift(schemeId: SchemeId): Future[Seq[Candidate]]
+  def siftApplicationForScheme(applicationId: String, result: SchemeEvaluationResult): Future[Unit]
+
+}
+
+class ApplicationSiftMongoRepository(
+  val dateTime: DateTimeFactory,
+  val siftableSchemes: Seq[Scheme]
+)(implicit mongo: () => DB)
+  extends ReactiveRepository[ApplicationForSift, BSONObjectID](CollectionNames.APPLICATION, mongo,
+    ApplicationForSift.applicationForSiftFormat,
+    ReactiveMongoFormats.objectIdFormats
+) with ApplicationSiftRepository {
+
+  val thisApplicationStatus = ApplicationStatus.SIFT
+  val prevPhase = ApplicationStatus.PHASE3_TESTS_PASSED_NOTIFIED
+  val prevTestGroup = "PHASE3"
+
+  val eligibleForSiftQuery = BSONDocument("$and" -> BSONArray(
+    BSONDocument("applicationStatus" -> prevPhase),
+    BSONDocument(s"testGroups.$prevTestGroup.evaluation.result" -> BSONDocument("$elemMatch" ->
+      BSONDocument("schemeId" -> BSONDocument("$in" -> siftableSchemes.map(_.id)),
+      "result" -> EvaluationResults.Green.toPassmark)
+  ))))
+
+  def nextApplicationsForSiftStage(batchSize: Int): Future[List[ApplicationForSift]] = {
+    selectRandom[BSONDocument](eligibleForSiftQuery, batchSize).map {
+      _.map { document =>
+        val applicationId = document.getAs[String]("applicationId").get
+        val testGroupsRoot = document.getAs[BSONDocument]("testGroups").get
+        val phase3PassMarks = testGroupsRoot.getAs[BSONDocument](prevTestGroup).get
+        val phase3Evaluation = phase3PassMarks.getAs[PassmarkEvaluation]("evaluation").get
+        ApplicationForSift(applicationId, phase3Evaluation)
+      }
+    }
+  }
+
+  def progressApplicationToSiftStage(application: ApplicationForSift): Future[Unit] = {
+    val query = BSONDocument("applicationId" -> application.applicationId) ++ eligibleForSiftQuery
+    val update = BSONDocument("applicationStatus" -> ApplicationStatus.SIFT)
```

Comment:
I'd set a matching progress status (and crucially the date) here so we know WHEN they moved to sift